pkg/pillar/containerd/containerd.go

```diff
@@ -338,6 +663,63 @@ func (client *Client) CtrMountSnapshot(ctx context.Context, snapshotID, targetPa
 	return mounts[0].Mount(targetPath)
 }
 
+// CtrCreateSnapshotFromExistingRootfs creates a snapshot from an existing rootfs
+func (client *Client) CtrCreateSnapshotFromExistingRootfs(ctx context.Context, snapshotName, rootfsPath string) error {
```

Comment:
creating a snapshot is much simpler than creating an image - you only package the files from rootfs, without worrying about the config or manifest. the logic here is different however from creating the image - here we don't create any tarballs, but simply create an empty snapshot - copy the files there - save the snapshot with files.

Reply:
what is the purpose of that? Usually, when ctrd creates a snapshot, it is based on existing filesystems, essentially layers from images. Does this bypass all of that? I had thought this process was:

1. Load rootfs as image into containerd
2. Use containerd to create container (which would handle all of the snapshots and all of that)

Where is this needed?

Reply:
This function is not used in the current implementation. I based my first draft on it, which just created a snapshot from rootfs instead of the whole xen-tools image and then reused this snapshot to create user apps. I decided to keep this function in case we might need it in the future.

Reply:
Sure :) I just wanted to save it because it's a working aproach and it took me a lot of time to develop it. And it might be useful in the future.

Reply:
That does make sense. Then can you explain a bit more when it is useful? snapshots from a rootfs don't make sense to me (in the context of containerd). It is usually, "load an image" and then "create snapshots from that image". Those functions (I think) are elsewhere, so is this something more than just a wrapper around those two?

Reply:
well in our particular scenario, we don't really need a full image, we just need the files from rootfs, since we create the OCI config ourselves. so actually a snapshot works well, because a snapshot from overlayfs-snapshotter is just a overlayfs mount of the image, where the image stays read-only and the changes only exist in the upper directory.

so go like this:
- create (prepare) a new empty snapshot called xen-tools, unless it already exists (same as with the image)
- copy the files from xen-tools rootfs on disk into that snapshot
- commit that snapshot and give it a label to prevent garbage collecting
- use that snapshot as a base to create (prepare) new snapshots for each user app container
- those new snapshots will be managed by containerd internally, so we don't have to take care of them